include/hip/hcc_detail/functional_grid_launch.hpp

```diff
@@ -155,30 +155,17 @@ void hipLaunchKernelGGLImpl(
     std::uint32_t sharedMemBytes,
     hipStream_t stream,
     void** kernarg) {
-    auto it0 = functions().find(function_address);
-
-    if (it0 == functions().cend()) {
-        hip_throw(std::runtime_error{
-            "No device code available for function: " +
-            name(function_address)});
-    }
 
     auto agent = target_agent(stream);
+    auto it = functions(agent).find(function_address);
```

Comment:
This is unsafe. If somehow the program invariants are broken and either agent contains rubbish or you're in a scenario where you have a surprising agent that was not part of the initial enumeration the results will be... interesting:)

Reply:
Should we check the agent is valid/known in the `functions()` function, rather than have the caller check for a known agent prior to calling `functions()`?